### DISCUSSION TITLE
Resolved: Green light to Stabilize `bech32`

### DISCUSSION THREAD

**Discussion by tcharding** [rust-bitcoin/rust-bitcoin#1128] [Announcements]
I emailed @clarkmoody and we have the green light to push the `bech32` crate towards stabilization. We can, hack, merge, release, as we see fit to get it across the line.

**Comment by Kixunil** [comment]
Oh, I got scared you mean to stabilize it as-is. There's still a shitload of work to do.

**Reply by apoelstra** [reply]
There are a number of things, but given how small and targeted the library I'm not sure it's a "shitload". My read is that we need to

* ~Update the MSRV~ done already: https://github.com/rust-bitcoin/rust-bech32/pull/57
* Then add `alloc` feature so that no-std will work: https://github.com/rust-bitcoin/rust-bech32/pull/71
* Then address your "wtf is going on with this API" issue

**Reply by Kixunil** [reply]
I consider it a lot, at least compared to what I expected before I actually looked into it. :)

**Comment by tcharding** [comment]
> Then add alloc feature so that no-std will work

I dug around (with my minimal no-std knowledge) for a while but it looks like `bech32` works with no-std but always needs an allocator. Can you explain further what you mean please?

**Reply by Kixunil** [reply]
I'm not sure what he meant but I'd like to see support for no-alloc by conditionally disabling APIs that need to allocate. So basically, just `u5` type and conversions using iterators.

**Reply by apoelstra** [reply]
I just meant https://github.com/rust-bitcoin/rust-bech32/pull/45

I didn't look at the PR but was a little surprised that we pushed a MSRV bump through without it.'

**Reply by Kixunil** [reply]
Yeah, that looks like what I meant although `arrayvec` and generics instead of `u5` look suspicious.

**Reply by tcharding** [reply]
Sweet, I'm on it.

**Comment by tcharding** [comment]
There are a bunch of open PRs on `rust-bech32` that I did last year, during the release of 0.30 since we are releasing everything else why don't we merge all that and release another version of `rust-bech32`?

**Reply by clarkmoody** [reply]
I like that idea.

**Comment by tcharding** [comment]
The stabalization plan is handled here: https://github.com/rust-bitcoin/rust-bech32/issues/60

**Comment by tcharding** [comment]
`bech32` is not in the public API of `primitives` or `units` so we have decided against working on stabalizing it right now.
